src/matbench_genmetrics/utils/match.py

```diff
@@ -166,35 +145,53 @@ def cdvae_cov_compstruct_match_matrix(
 ALLOWED_MATCH_TYPES = ["StructureMatcher", "cdvae_coverage"]
 
 
-def get_match_matrix(
+def get_structure_match_matrix(
     test_structures,
     gen_structures,
     match_type="cdvae_coverage",
     symmetric=False,
     verbose=False,
     **match_kwargs,
 ):
-    assert (
-        match_type in ALLOWED_MATCH_TYPES
-    ), f"type must be one of {ALLOWED_MATCH_TYPES}"
-
-    if match_type == "cdvae_coverage":
-        return cdvae_cov_compstruct_match_matrix(
+    if match_type == "StructureMatcher":
+        return structure_pairwise_match_matrix(
             test_structures,
             gen_structures,
+            match_type="StructureMatcher",
             symmetric=symmetric,
             verbose=verbose,
             **match_kwargs,
         )
-    elif match_type == "StructureMatcher":
-        return structure_pairwise_match_matrix(
-            test_structures,
-            gen_structures,
-            match_type="StructureMatcher",
+    else:
+        raise ValueError(
+            f"Unknown match type {match_type}. Must be one of {ALLOWED_MATCH_TYPES}"
+        )  # noqa: E501
+
+
+def get_fingerprint_match_matrix(
+    test_comp_fingerprints,
+    gen_comp_fingerprints,
+    test_struct_fingerprints,
+    gen_struct_fingerprints,
+    match_type="cdvae_coverage",
+    symmetric=False,
+    verbose=False,
+    **match_kwargs,
+):
+    if match_type == "cdvae_coverage":
```

Comment:
if you were to add the hashes, it would be something like: 
```python
from structuregraph_helpers.cli import create_hashes_for_structure
import concurrent.futures
# get an OrderedDict of different hash "flavors"

all_hashes = []

with concurrent.futures.ProcessPoolExecutor(max_workers=n_jobs) as executor: 
	for hashes, name in zip(executor.map(create_hashes_for_structure, structures), names)
		# the identifier "name" might also just be an index. However, it is good to have one 
		# with this multiprocessing stuff
		hashes['name'] = name
		all_hashes.append(hashes)
```

Reply:
https://github.com/sparks-baird/matbench-genmetrics/issues/21

Reply:
For the fingerprinting, do you think this would speed things up overall? (given that `CrystalNN` is used for the structural fingerprints atm). For `StructureMatcher`, I think it makes sense.

Reply:
if you want to optimize for speed, I'd go with a cutoff-based method for computing the graph (e.g. the `CutoffDictNN` with the vesta presets). The voronoi tesselations tend to be slow for large structures